pages/_app.tsx

```diff
@@ -20,18 +78,6 @@ export default class MyApp extends App {
 	render(): JSX.Element {
 		const { Component, pageProps } = this.props
 
-		return (
-			<>
-				<Head>
-					{/* https://err.sh/next.js/no-document-title */}
-					<title>My new cool app</title>
-				</Head>
-				<ThemeProvider theme={theme}>
-					{/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
-					<CssBaseline />
-					<Component {...pageProps} />
-				</ThemeProvider>
-			</>
-		)
+		return <AppWithHooks component={Component} pageProps={pageProps} />
```

Comment:
AppWithHooks feels redundant here. Why can't the MyApp class itself just implement that logic? If it does need to be its own component then i'd prefer it also be its own module in components/.

AFAICT you are literally just accessing some props, so why not access them in MyApp instead of passing them into AppWithHooks?

Reply:
The props are being inherited from the Next App class, this isn't possible with functional components

Reply:
But it isn't a functional component. it's a classical component. so why can't it use its inherited props? Why does it need to pass them to some random functional component?

```jsx
// psudocode
render(): JSX.Element {
    const [themeType, setThemeType] = useState<'dark' | 'light'>('dark')
	const [primary, setPrimary] = useState<Color>(blue)
	const [secondary, setSecondary] = useState<Color>(deepPurple)
	const theme = getTheme(themeType)

    return <>
			<Head>
				{/* https://err.sh/next.js/no-document-title */}
				<title>Impact</title>
			</Head>
			<ThemeProvider theme={theme}>
				{/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
				<CssBaseline />
				<AppContext.Provider
					value={{
						setThemeType,
						setPrimary,
						setSecondary,
						themeType,
						themePrimary: primary,
						themeSecondary: secondary,
					}}
				>
					<props.Component {...props.pageProps} />
				</AppContext.Provider>
			</ThemeProvider>
		</>
}
```

Reply:
useState and context uses hooks. That means that class components simply do not support this kind of syntax. That's why a functional component is created.

Reply:
Also the code you sent me would clear the component state every render :joy:

Reply:
> useState and context uses hooks. That means that class components simply do not support this kind of syntax. That's why a functional component is created.

This sounds backwards; [Hooks are a new addition in React 16.8. They let you use state and other React features without writing a class](https://reactjs.org/docs/hooks-faq.html). If hooks are a way to use react features _without_ writing a class, then surely a class is another way to do the same things.



> the code you sent me would clear the component state every render

It was just an example pasted together on my phone to illustrate what I was saying. Hence the `//psudocode` comment